src/game/game.c

```diff
@@ -53,7 +53,11 @@ int32_t StopGame()
     if (InvExtraData[0] == 0) {
         return GF_START_SAVED_GAME | InvExtraData[1];
     } else if (InvExtraData[0] == 1) {
-        return GF_START_GAME | GF.first_level_num;
+        if (InvMode == INV_DEATH_MODE) {
+            return GF_START_GAME | CurrentLevel;
+        } else {
```

Comment:
I'd reword it to `return GF_START_GAME | (InvMode == INV_DEATH_MODE ? CurrentLevel : GF.first_level_num);`